nixos/modules/services/databases/postgresql.nix

```diff
@@ -391,12 +391,11 @@ in
       };
 
       extraPlugins = mkOption {
-        type = types.listOf types.path;
-        default = [];
-        example = literalExpression "with pkgs.postgresql_15.pkgs; [ postgis pg_repack ]";
+        type = with types; coercedTo (listOf path) (path: _ignorePg: path) (functionTo (listOf path));
```

Comment:
```suggestion
        type = with types; coercedTo (listOf path) (path: _ignorePg: lib.warnIf (lib.isInOldestRelease 2405) "The postgresql module passes an attribute set of postgresql extension packages to the services.postgresql.extraPlugins option. Please write it as a function to a list of packages. The plain list of packages syntax will be removed." path) (functionTo (listOf path));
```

It really is only syntax though. Users can still write `don'tCare: [ nixpkgs2205pinned.postgresqlPackages.bad ]`.

Reply:
With the warning we can at least nudge them into doing the right thing.

Reply:
> It really is only syntax though. Users can still write don'tCare: [ nixpkgs2205pinned.postgresqlPackages.bad ].

That's true, but by naming the argument `don'tCare` (it's kinda cool that we can do single quotes in identifiers for that btw ;-)), folks explicitly stated that they - well - don't care, but are aware of the risks (hopefully).

If you use a list because you missed the part in the release-notes and are not aware of the issue, that's more of a problem IMHO.

I'm fine with a warning as well, though.